### PR TITLE
feat: Make async session statuses describe a session transition inste…

### PR DESCRIPTION
**Pull request by kpears201**
…ad of a session

### CODE REVIEW

**Comment by kpears201** [comment]
The terms "canceled session" and "created session" did not work for the use case where an app was preloaded into the inactive state and then session is called to bring it to foreground but is gated by a user grant. The session was already returned in a previous session call, so it does not make sense to call onSessionCreated for a session that was already created. Also, if the user grant is deferred, then onSessionCanceled does not make sense in this case. The session will still exist in the inactive state, so the session was not "canceled".
This change is so that the event describes a "transition" of a session instead of the session itself. For the use case above, when the session API is called, a transition is started from the inactive status to the active status. This transition can then "completed" or be "canceled". If a transition is canceled, then it just remains in the current status (inactive).
The "state" field was also dropped and replaced with a boolean "transitionPending". This makes it more of a transitive property instead of a final state. A transition can be pending for when the status is going to "not-exists" to "active" or from "inactive" to "active". What the caller cares about is if a transition is happening as a result of the session call so they know to check the events to get the updated session asynchronously.

File: src/openrpc/lifecycle_management.json [hunk 1]
```diff
@@ -64,7 +64,7 @@
...
+            "summary": "Called before an app is launched in order to set up the lifecycle session. The session should be given an intent that will be passed to the application. This will behave differently depending on the current state of the application. If the application is already loaded but not active then a new active session is started. The intent is given to the application via the onNavigateTo event. If the app is not loaded, then it is considered a 'cold start'. A new loaded session will be created and the intent will be stored for the app to access via Parameters.initialization. Before a session is created for the application some checks may take place. For instance, if a user grant is required for the application then the user grant will be resolved before the session is transitions to the active status. In this case, this API will return immediately with a transitionPending=true flag. The updated session will then be sent as an event in the LifecycleManagement.onSessionTransitionCompleted event. If the pre-session checks fail, such as a user deferring a grant, then the LifecycleManagement.onSessionTransitionCanceled event will be emitted.",
```
Comment (Vinodsathyaseelan) [inline src/openrpc/lifecycle_management.json:67]: What does pre-session checks fail mean ?  
User is not responding to the acknowledgement challenge (challenge request timeout) or ignore the request ? 
Allow or Deny for the challenge is a valid pre-check response. right ?

**Review by Vinodsathyaseelan — APPROVED** [review]

File: src/openrpc/lifecycle_management.json [hunk 2]
```diff
@@ -419,15 +402,15 @@
                             "sessionId": "c0d012b4-0854-4bbc-82d5-ab52ac29dd94",
```
Comment (Vinodsathyaseelan) [inline src/openrpc/lifecycle_management.json:402]: appId is missing in this response.

File: src/openrpc/lifecycle_management.json (hunk 1 above)
Comment (kpears201) [reply src/openrpc/lifecycle_management.json:67]: yes, only null grant response is a pre check fail right now

File: src/openrpc/lifecycle_management.json (hunk 2 above)
Comment (kpears201) [reply src/openrpc/lifecycle_management.json:402]: it is there, did you expand the up in what github shows?

**Pull request merged**